swupd/swupd.go

```diff
@@ -378,6 +411,29 @@ func (s *SoftwareUpdater) VerifyWithBundles(version string, mirror string, bundl
 	return nil
 }
 
+// DownloadBundles downloads the bundle list to the OfflineContentDir within the installer image
+func (s SoftwareUpdater) DownloadBundles(version, mirror string, bundles []string) error {
+	var err error
+
+	s.downloadOnly = true
+
+	// The statedir-cache is set to the statedir that was used to install bundles to the
+	// installer image which reduces the number of bundle downloads. The statedir specifies the
+	// location to download offline content which is set ot the offline content directory within
+	// the installer image.
+	s.stateDirCache = s.stateDir
+	s.stateDir = filepath.Join(s.rootDir, conf.OfflineContentDir)
+
+	// Set swupd's --path argument to an empty directory so that all listed bundles will be
+	// downloaded (Swupd skips over previously installed bundles)
+	if s.rootDir, err = ioutil.TempDir("", "installerTmp-"); err != nil {
+		return err
+	}
+	defer func() { _ = os.RemoveAll(s.rootDir) }()
+
+	return s.VerifyWithBundles(version, mirror, bundles)
```

Comment:
Now, if you download offline content and are generating an ISO, we call `VerifyWithBundles` three times -- the output is nearly identical each time and could be misleading to the user.

Please add a new parameter to  `VerifyWithBundles` with a prefix string for all of the steps. For the main OS install we can use a prefix like 'Target OS', for the offline content 'Offline Content', and for the ISO, 'ISO Boot OS'.